Failed to compile on Fedora 40

```
$ uname -ar
Linux shrimp-paste 6.8.11-300.fc40.ppc64le #1 SMP Mon May 27 14:48:15 UTC 2024 ppc64le GNU/Linux

$ gcc -v
Using built-in specs.
COLLECT_GCC=/usr/bin/gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/ppc64le-redhat-linux/14/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: ppc64le-redhat-linux
Configured with: ../configure --enable-bootstrap --enable-languages=c,c++,fortran,objc,obj-c++,ada,go,m2,lto --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-shared --enable-threads=posix --enable-checking=release --enable-targets=powerpcle-linux --disable-multilib --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-gcc-major-version-only --enable-libstdcxx-backtrace --with-libstdcxx-zoneinfo=/usr/share/zoneinfo --with-linker-hash-style=gnu --enable-plugin --enable-initfini-array --with-isl=/builddir/build/BUILD/gcc-14.1.1-20240522/obj-ppc64le-redhat-linux/isl-install --enable-offload-targets=nvptx-none --enable-offload-defaulted --without-cuda-driver --enable-gnu-indirect-function --enable-secureplt --with-long-double-128 --with-long-double-format=ieee --with-cpu-32=power8 --with-tune-32=power8 --with-cpu-64=power8 --with-tune-64=power8 --build=ppc64le-redhat-linux --with-build-config=bootstrap-lto --enable-link-serialization=1
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 14.1.1 20240522 (Red Hat 14.1.1-4) (GCC)
tle@shrimp-paste:~/Work/afni/build$ g++ -v
Using built-in specs.
COLLECT_GCC=/usr/bin/g++
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/ppc64le-redhat-linux/14/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: ppc64le-redhat-linux
Configured with: ../configure --enable-bootstrap --enable-languages=c,c++,fortran,objc,obj-c++,ada,go,m2,lto --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-shared --enable-threads=posix --enable-checking=release --enable-targets=powerpcle-linux --disable-multilib --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-gcc-major-version-only --enable-libstdcxx-backtrace --with-libstdcxx-zoneinfo=/usr/share/zoneinfo --with-linker-hash-style=gnu --enable-plugin --enable-initfini-array --with-isl=/builddir/build/BUILD/gcc-14.1.1-20240522/obj-ppc64le-redhat-linux/isl-install --enable-offload-targets=nvptx-none --enable-offload-defaulted --without-cuda-driver --enable-gnu-indirect-function --enable-secureplt --with-long-double-128 --with-long-double-format=ieee --with-cpu-32=power8 --with-tune-32=power8 --with-cpu-64=power8 --with-tune-64=power8 --build=ppc64le-redhat-linux --with-build-config=bootstrap-lto --enable-link-serialization=1
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 14.1.1 20240522 (Red Hat 14.1.1-4) (GCC)

$ mkdir build && cd build && cmake .. && make -j384

[ 19%] Building C object src/CMakeFiles/mri_objs.dir/mri_genARMA11.c.o
[ 19%] Building C object src/CMakeFiles/mri_objs.dir/mri_get_cmass.c.o
/home/tle/Work/afni/src/mri_dicom_hdr.c: In function ‘exportData’:
/home/tle/Work/afni/src/mri_dicom_hdr.c:4694:21: error: implicit declaration of function ‘swab’ [-Wimplicit-function-declaration]
 4694 |                     swab(p, b, length);
      |                     ^~~~
make[2]: *** [src/CMakeFiles/mri_objs.dir/build.make:440: src/CMakeFiles/mri_objs.dir/mri_dicom_hdr.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 19%] Building C object src/CMakeFiles/mri_objs.dir/mri_histobyte.c.o
make[1]: *** [CMakeFiles/Makefile2:1910: src/SUMA/GLUT/libglut/CMakeFiles/GLUT.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 20%] Linking C executable ../../../targets_built/nifti_tester001
[ 20%] Built target nifti_tester001
[ 20%] Linking C executable ../../../targets_built/nifti_first_test_program
[ 20%] Built target nifti_first_test_program
[ 20%] Linking C executable ../../targets_built/file_tool
[ 20%] Built target file_tool
[ 20%] Linking C executable ../../../targets_built/nifti1_tool
[ 20%] Built target nifti1_tool
[ 20%] Linking C shared library ../../../targets_built/libnifticdf.so
[ 20%] Built target nifticdf
[ 20%] Linking C shared library ../../targets_built/libgiftiio.so
[ 20%] Built target giftiio
make[1]: *** [CMakeFiles/Makefile2:3462: src/CMakeFiles/mri_objs.dir/all] Error 2
[ 20%] Linking C executable ../../../targets_built/nifti_tool
[ 20%] Built target nifti_tool
make: *** [Makefile:136: all] Error 2

/home/tle/Work/afni/src/imseq.c: In function ‘open_MCW_imseq’:
/home/tle/Work/afni/src/imseq.c:1257:26: error: passing argument 4 of ‘XtInsertEventHandler’ from incompatible pointer type [-Wincompatible-pointer-types]
 1257 |                          ISQ_drawing_EV ,       /* super-handler! */
      |                          ^~~~~~~~~~~~~~
      |                          |
      |                          void (*)(struct _WidgetRec *, void *, XEvent *, RwcBoolean *) {aka void (*)(struct _WidgetRec *, void *, XEvent *, unsigned char *)}
In file included from /home/tle/Work/afni/src/imseq.h:16,
                 from /home/tle/Work/afni/src/imseq.c:96:
/usr/include/X11/Intrinsic.h:886:5: note: expected ‘XtEventHandler’ {aka ‘void (*)(struct _WidgetRec *, void *, XEvent *, char *)’} but argument is of type ‘void (*)(struct _WidgetRec *, void *, XEvent *, RwcBoolean *)’ {aka ‘void (*)(struct _WidgetRec *, void *, XEvent *, unsigned char *)’}
  886 |     XtEventHandler      /* proc */,
      |     ^~~~~~~~~~~~~~
/home/tle/Work/afni/src/imseq.c:1330:28: error: passing argument 4 of ‘XtInsertEventHandler’ from incompatible pointer type [-Wincompatible-pointer-types]
 1330 |                            ISQ_butsave_EV ,     /* handler */
      |                            ^~~~~~~~~~~~~~
      |                            |
      |                            void (*)(struct _WidgetRec *, void *, XEvent *, RwcBoolean *) {aka void (*)(struct _WidgetRec *, void *, XEvent *, unsigned char *)}
/usr/include/X11/Intrinsic.h:886:5: note: expected ‘XtEventHandler’ {aka ‘void (*)(struct _WidgetRec *, void *, XEvent *, char *)’} but argument is of type ‘void (*)(struct _WidgetRec *, void *, XEvent *, RwcBoolean *)’ {aka ‘void (*)(struct _WidgetRec *, void *, XEvent *, unsigned char *)’}
  886 |     XtEventHandler      /* proc */,
      |     ^~~~~~~~~~~~~~
/home/tle/Work/afni/src/imseq.c:1340:26: error: passing argument 4 of ‘XtInsertEventHandler’ from incompatible pointer type [-Wincompatible-pointer-types]
 1340 |                          ISQ_butdisp_EV ,     /* handler */
      |                          ^~~~~~~~~~~~~~
      |                          |
      |                          void (*)(struct _WidgetRec *, void *, XEvent *, RwcBoolean *) {aka void (*)(struct _WidgetRec *, void *, XEvent *, unsigned char *)}
/usr/include/X11/Intrinsic.h:886:5: note: expected ‘XtEventHandler’ {aka ‘void (*)(struct _WidgetRec *, void *, XEvent *, char *)’} but argument is of type ‘void (*)(struct _WidgetRec *, void *, XEvent *, RwcBoolean *)’ {aka ‘void (*)(struct _WidgetRec *, void *, XEvent *, unsigned char *)’}
  886 |     XtEventHandler      /* proc */,
      |     ^~~~~~~~~~~~~~
/home/tle/Work/afni/src/imseq.c:1655:28: error: passing argument 4 of ‘XtInsertEventHandler’ from incompatible pointer type [-Wincompatible-pointer-types]
 1655 |                            ISQ_butcrop_EV ,     /* handler */
      |                            ^~~~~~~~~~~~~~
      |                            |
      |                            void (*)(struct _WidgetRec *, void *, XEvent *, RwcBoolean *) {aka void (*)(struct _WidgetRec *, void *, XEvent *, unsigned char *)}
/usr/include/X11/Intrinsic.h:886:5: note: expected ‘XtEventHandler’ {aka ‘void (*)(struct _WidgetRec *, void *, XEvent *, char *)’} but argument is of type ‘void (*)(struct _WidgetRec *, void *, XEvent *, RwcBoolean *)’ {aka ‘void (*)(struct _WidgetRec *, void *, XEvent *, unsigned char *)’}
  886 |     XtEventHandler      /* proc */,
      |     ^~~~~~~~~~~~~~
/home/tle/Work/afni/src/imseq.c:1812:26: error: passing argument 4 of ‘XtInsertEventHandler’ from incompatible pointer type [-Wincompatible-pointer-types]
 1812 |                          ISQ_drawing_EV ,        /* super-handler! */
      |                          ^~~~~~~~~~~~~~
      |                          |
      |                          void (*)(struct _WidgetRec *, void *, XEvent *, RwcBoolean *) {aka void (*)(struct _WidgetRec *, void *, XEvent *, unsigned char *)}
/usr/include/X11/Intrinsic.h:886:5: note: expected ‘XtEventHandler’ {aka ‘void (*)(struct _WidgetRec *, void *, XEvent *, char *)’} but argument is of type ‘void (*)(struct _WidgetRec *, void *, XEvent *, RwcBoolean *)’ {aka ‘void (*)(struct _WidgetRec *, void *, XEvent *, unsigned char *)’}
  886 |     XtEventHandler      /* proc */,
      |     ^~~~~~~~~~~~~~
/home/tle/Work/afni/src/imseq.c: In function ‘drive_MCW_imseq’:
/home/tle/Work/afni/src/imseq.c:8881:15: error: passing argument 4 of ‘XtInsertEventHandler’ from incompatible pointer type [-Wincompatible-pointer-types]
 8881 |               ISQ_button2_EV ,      /* handler routine */
      |               ^~~~~~~~~~~~~~
      |               |
      |               void (*)(struct _WidgetRec *, void *, XEvent *, RwcBoolean *) {aka void (*)(struct _WidgetRec *, void *, XEvent *, unsigned char *)}
/usr/include/X11/Intrinsic.h:886:5: note: expected ‘XtEventHandler’ {aka ‘void (*)(struct _WidgetRec *, void *, XEvent *, char *)’} but argument is of type ‘void (*)(struct _WidgetRec *, void *, XEvent *, RwcBoolean *)’ {aka ‘void (*)(struct _WidgetRec *, void *, XEvent *, unsigned char *)’}
  886 |     XtEventHandler      /* proc */,
      |     ^~~~~~~~~~~~~~
/home/tle/Work/afni/src/imseq.c:8905:15: error: passing argument 4 of ‘XtRemoveEventHandler’ from incompatible pointer type [-Wincompatible-pointer-types]
 8905 |               ISQ_button2_EV ,      /* handler routine */
      |               ^~~~~~~~~~~~~~
      |               |
      |               void (*)(struct _WidgetRec *, void *, XEvent *, RwcBoolean *) {aka void (*)(struct _WidgetRec *, void *, XEvent *, unsigned char *)}
/usr/include/X11/Intrinsic.h:862:5: note: expected ‘XtEventHandler’ {aka ‘void (*)(struct _WidgetRec *, void *, XEvent *, char *)’} but argument is of type ‘void (*)(struct _WidgetRec *, void *, XEvent *, RwcBoolean *)’ {aka ‘void (*)(struct _WidgetRec *, void *, XEvent *, unsigned char *)’}
  862 |     XtEventHandler      /* proc */,
      |     ^~~~~~~~~~~~~~
      
```

Any advice to workaround this issue is greatly appreciated

--- a/src/mri_dicom_hdr.c
+++ b/src/mri_dicom_hdr.c
@@ -66,6 +66,11 @@
 #include <stdarg.h>
 #include <ctype.h>
 
+#if defined(LINUX) || defined(LINUX2)
+#define __USE_XOPEN
+#include <unistd.h>
+#endif
+	
 #include "mri_dicom_hdr.h"
 #include "mri_dicom_elist.h"  /* for whining  [6 Mar 2019 rickr] */
 

